core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt

```diff
@@ -20,32 +22,35 @@ class MovementSystem(
 	priority
 ) {
 
-	private lateinit var entities : ImmutableArray<Entity>
+	private lateinit var entities: ImmutableArray<Entity>
 	private val movementMapper = mapperFor<MovementComponent>()
 	private val transformMapper = mapperFor<TransformComponent>()
 
-	override fun addedToEngine ( engine : Engine ) {
+	override fun addedToEngine(engine: Engine) {
 		entities = engine.getEntitiesFor(
 			Family.all(MovementComponent::class.java, TransformComponent::class.java).get()
 		)
 	}
 
-	override fun update ( deltaTime : Float ) {
+	override fun update(deltaTime: Float) {
 
-		for ( entity in entities ) {
+		for (entity in entities) {
 			val movement = movementMapper.get(entity)
 			val transform = transformMapper.get(entity)
 
 			// If move command, accelerate (also, this is demo stuff, so adapt or remove)
 			movement.velocity.x += movement.acceleration * deltaTime * (movement.maxSpeed - movement.velocity.len())
 			// If not decelerate
 
-			transform.rotation += 10 * deltaTime // remove
+			when {
+				movement.moveDown -> transform.position = move(Input.Keys.DOWN, transform.position)
+				movement.moveUp -> transform.position = move(Input.Keys.UP, transform.position)
+				movement.moveLeft -> transform.position = move(Input.Keys.LEFT, transform.position)
+				movement.moveRight -> transform.position = move(Input.Keys.RIGHT, transform.position)
+			}
 
-			transform.position.x += movement.velocity.x * deltaTime
-			transform.position.y += movement.velocity.y * deltaTime
+			if (transform.position.x > viewport.worldWidth / 2) transform.position.copy(x = -viewport.worldWidth / 2) // remove
 
-			if (transform.position.x > viewport.worldWidth / 2) transform.position.x = -viewport.worldWidth / 2 // remove
 		}
```

Comment:
Are we loosing the main functionality of the Movement System here (move entities based on movement component)? Is this implementation trying to accomplish the Command Pattern? (I ask just to know) Because another way to implement this could be to make modify the velocity of the entity by the input keys. Didn't you do that because of the command pattern?

Reply:
I first tried changing the velocity on input, but it felt kinda weird to control since there is a bit of a delay right when moving the player with acceleration and deceleration so it feels smoother when you just update the position directly (which technically means we can get rid of the `MovementComponent`)

Reply:
And what if we instead get rid of acceleration/deceleration instead of the MovementComponent entirely? If we ignore the acceleration mechanics, it should be smooth as your current implementation, right? I feel like we are loosing something here, but not sure and I don't wont to block you without specific reasons, you know. So if you think it is good enough I'll trust you, just say it and I'll approve the change. We can come back to this later if we have issue, I think is better to try and make error than talk too much! :)